src/ethers.ts

```diff
@@ -176,9 +170,28 @@ export class EthersMulticall {
       };
 
       // Overwrite the function with a dataloader batched call
-      Object.defineProperty(copy, fragment.name, descriptor);
-      Object.defineProperty(copy.callStatic, fragment.name, descriptor);
-      Object.defineProperty(copy.functions, fragment.name, descriptor);
+      Object.defineProperty(copy, property, descriptor);
+      Object.defineProperty(copy.callStatic, property, descriptor);
+      Object.defineProperty(copy.functions, property, descriptor);
+    };
+
+    const uniqueNames: { [name: string]: FunctionFragment[] } = {};
+
+    Object.entries(contract.interface.functions)
+      .filter(([, fragment]) => ["view", "pure"].includes(fragment.stateMutability))
+      .forEach(([signature, fragment]) => {
```

Comment:
can be merged to only one `forEach` loop with an early return if `!["view", "pure"].includes(fragment.stateMutability)`